### PR TITLE
feat: endpoint to check for transactions in mempool

### LINKED ISSUE #85: Create GET endpoint to check for transactions
Create an endpoint to check if a given transaction hash is in the Cardano node's mempool.

### PR DESCRIPTION
**Pull request by wolf31o2**
Closes #85

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,15 +3,15 @@ module github.com/blinklabs-io/tx-submit-api
...
+	github.com/blinklabs-io/gouroboros v0.41.0
 	github.com/fxamacker/cbor/v2 v2.4.0
 	github.com/gin-contrib/zap v0.1.0
 	github.com/gin-gonic/gin v1.9.0
 	github.com/kelseyhightower/envconfig v1.4.0
 	github.com/penglongli/gin-metrics v0.1.10
 	github.com/swaggo/files v1.0.1
 	github.com/swaggo/gin-swagger v1.6.0
-	github.com/swaggo/swag v1.16.1
+	github.com/swaggo/swag v1.8.12
```
Comment (agaffney) [inline go.mod:14]: Why downgrade this?

File: internal/api/api.go [hunk 2]
```diff
@@ -120,7 +121,95 @@ func handleHealthcheck(c *gin.Context) {
...
+		}
+	}()
+	defer func() {
+		// Close Ouroboros connection
+		oConn.Close()
+	}()
+	hasTx, err := oConn.LocalTxMonitor().Client.HasTx(cborData)
+	if err != nil {
+		logger.Errorf("failure getting transaction: %s", err)
+		c.JSON(400, fmt.Sprintf("failure getting transaction: %s", err))
```
Comment (agaffney) [inline internal/api/api.go:202]: Should this be a 500?

File: internal/api/api.go [hunk 3]
```diff
@@ -120,7 +121,95 @@ func handleHealthcheck(c *gin.Context) {
...
+		c.JSON(400, fmt.Sprintf("invalid transaction hash: %s", err))
+		return
+	}
+
+	txHash := uriParams.TxHash
+	// convert to cbor bytes
+	cborData, err := cbor.Marshal(txHash)
+	if err != nil {
+		logger.Errorf("failed to encode transaction hash to CBOR: %s", err)
+		c.JSON(400, fmt.Sprintf("failed to encode transaction hash to CBOR: %s", err))
```
Comment (agaffney) [inline internal/api/api.go:158]: Should this be a 500?

File: go.mod (hunk 1 above)
Comment (wolf31o2) [reply go.mod:14]: I can't find that 1.16.1 on the swag GitHub, so I was unable to upgrade locally.

File: internal/api/api.go (hunk 2 above)
Comment (wolf31o2) [reply internal/api/api.go:202]: Probably. I'll change it.

File: internal/api/api.go (hunk 3 above)
Comment (wolf31o2) [reply internal/api/api.go:158]: I think no. Here's we're encoding the path given by the user, so a failure is always going to be because of bad input.

**Review by agaffney — APPROVED** [review]

**Pull request merged**